django/contrib/admin/filters.py

```diff
@@ -153,26 +153,28 @@ def register(cls, test, list_filter_class, take_priority=False):
             cls._field_list_filters.append((test, list_filter_class))
 
     @classmethod
-    def create(cls, field, request, params, model, model_admin, field_path):
+    def create(cls, field, request, params, model, model_admin, field_path, title=None):
         for test, list_filter_class in cls._field_list_filters:
             if test(field):
-                return list_filter_class(field, request, params, model, model_admin, field_path=field_path)
+                return list_filter_class(
+                    field, request, params, model, model_admin, field_path=field_path, title=title
+                )
 
 
 class RelatedFieldListFilter(FieldListFilter):
-    def __init__(self, field, request, params, model, model_admin, field_path):
+    def __init__(self, field, request, params, model, model_admin, field_path, title=None):
         other_model = get_model_from_relation(field)
         self.lookup_kwarg = '%s__%s__exact' % (field_path, field.target_field.name)
         self.lookup_kwarg_isnull = '%s__isnull' % field_path
         self.lookup_val = params.get(self.lookup_kwarg)
         self.lookup_val_isnull = params.get(self.lookup_kwarg_isnull)
-        super().__init__(field, request, params, model, model_admin, field_path)
+        super().__init__(field, request, params, model, model_admin, field_path, title=title)
         self.lookup_choices = self.field_choices(field, request, model_admin)
         if hasattr(field, 'verbose_name'):
             self.lookup_title = field.verbose_name
         else:
             self.lookup_title = other_model._meta.verbose_name
-        self.title = self.lookup_title
+        self.title = title or self.lookup_title
```

Comment:
WDYT about adding `or self.title` here?


``` python
self.title = title or self.title or self.lookup_title
```

This would add support for the following:

``` python

class MyStaffFilter(BooleanFieldListFilter):
    title = “By can haz admin”  # no effect :(  overwritten in __init__

@admin.register(User)
class UserAdmin(UserAdminBase):
    list_filter = (
        ("is_staff", MyStaffFilter),
    )
```

Reply:
(actually meant to comment this under `FieldListFilter` class, but applies here too)